src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -152,18 +154,25 @@ void activity_classifier::train(gl_sframe data, std::string target_column_name,
 {
   // Begin printing progress.
   // TODO: Make progress printing optional.
-  
-  // TODO: Support validation set.
-  training_table_printer_.reset(new table_printer(
+
+  if (variant_is<gl_sframe>(validation_data) ){
+
+    training_table_printer_.reset(new table_printer(
+    { {"Iteration", 12}, {"Train Accuracy", 12}, {"Train Loss", 12}, {"Validation Accuracy", 12}, {"Validation Loss", 12}, {"Elapsed Time", 12} }));
+
+  }
+  else {
+
+    training_table_printer_.reset(new table_printer(
     { {"Iteration", 12}, {"Train Accuracy", 12}, {"Train Loss", 12}, {"Elapsed Time", 12} }));
-  
-  
-  // Instantiate the training dependencies: data iterator, compute context,
-  // backend NN model.
+  }
+
   init_train(std::move(data), std::move(target_column_name),
              std::move(session_id_column_name), std::move(validation_data),
              std::move(opts));
 
+  // Instantiate the training dependencies: data iterator, compute context,
```

Comment:
Why move this comment down here? It's meant to document the `init_train` invocation